File: frontend/src/api/mei.js
import client from './client'

const BASE = '/mei'

export const meiApi = {
  listar: (params) => client.get(BASE, { params }).then((r) => r.data),
  obter: (id) => client.get(`${BASE}/${id}`).then((r) => r.data),
  criar: (data) => client.post(BASE, data).then((r) => r.data),
  atualizar: (id, data) => client.put(`${BASE}/${id}`, data).then((r) => r.data),
  deletar: (id) => client.delete(`${BASE}/${id}`),
  resumoMensal: (ano) => client.get(`${BASE}/resumo/mensal`, { params: { ano } }).then((r) => r.data),
  resumoAnual: (ano) => client.get(`${BASE}/resumo/anual`, { params: { ano } }).then((r) => r.data),
}

File: backend/app/routers/mei.py
from typing import Optional, List

from fastapi import APIRouter, Depends, HTTPException, Query
from sqlalchemy.orm import Session

from app.database import get_db
from app.schemas.mei import (
    MEICreate,
    MEIUpdate,
    MEIResponse,
    MEIListResponse,
    MEIResumoMensal,
    MEIResumoAnual,
)
from app.services import mei_service

router = APIRouter()


@router.get("", response_model=MEIListResponse)
def listar_mei(
    mes: Optional[str] = None,
    ano: Optional[int] = None,
    page: int = Query(1, ge=1),
    per_page: int = Query(20, ge=1, le=100),
    db: Session = Depends(get_db),
):
    return mei_service.listar_mei(db, mes, ano, page, per_page)


@router.get("/resumo/mensal", response_model=List[MEIResumoMensal])
def resumo_mensal(ano: int, db: Session = Depends(get_db)):
    return mei_service.resumo_mensal(db, ano)


@router.get("/resumo/anual", response_model=MEIResumoAnual)
def resumo_anual(ano: int, db: Session = Depends(get_db)):
    return mei_service.resumo_anual(db, ano)


@router.get("/{mei_id}", response_model=MEIResponse)
def obter_mei(mei_id: int, db: Session = Depends(get_db)):
    lancamento = mei_service.obter_mei(db, mei_id)
    if not lancamento:
        raise HTTPException(status_code=404, detail="Lançamento MEI não encontrado")
    return lancamento


@router.post("", response_model=MEIResponse, status_code=201)
def criar_mei(data_in: MEICreate, db: Session = Depends(get_db)):
    return mei_service.criar_mei(db, data_in)


@router.put("/{mei_id}", response_model=MEIResponse)
def atualizar_mei(mei_id: int, data_in: MEIUpdate, db: Session = Depends(get_db)):
    lancamento = mei_service.atualizar_mei(db, mei_id, data_in)
    if not lancamento:
        raise HTTPException(status_code=404, detail="Lançamento MEI não encontrado")
    return lancamento


@router.delete("/{mei_id}", status_code=204)
def deletar_mei(mei_id: int, db: Session = Depends(get_db)):
    if not mei_service.deletar_mei(db, mei_id):
        raise HTTPException(status_code=404, detail="Lançamento MEI não encontrado")
